tests/Unit/Domain/Test_InterfaceHelpers.cpp

```diff
@@ -16,36 +16,81 @@ struct SomeNumber : db::SimpleTag {
-  const auto computed_number_on_interfaces = interface_apply<
-      ::Tags::InternalDirections<Dim>, tmpl::list<SomeNumber>, tmpl::list<>>(
-      [](const double& some_number_on_interface, const double factor) noexcept {
-        return factor * some_number_on_interface;
-      },
-      box, 2.);
-  CHECK(computed_number_on_interfaces.at(Direction<Dim>::upper_xi()) == 4.);
+  const auto box =
+      db::create<db::AddSimpleTags<::Tags::Element<Dim>,
+                                   ::Tags::Interface<DirectionsTag, SomeNumber>,
+                                   SomeVolumeArgument>,
+                 db::AddComputeTags<DirectionsTag>>(element,
+                                                    number_on_interfaces, 1.);
+  // Test applying a function to the interface and give an example
+  /// [interface_apply_example]
+  const auto computed_number_on_interfaces =
+      interface_apply<DirectionsTag, tmpl::list<SomeNumber, SomeVolumeArgument>,
+                      tmpl::list<SomeVolumeArgument>>(
+          [](const double& some_number_on_interface,
+             const double& volume_argument, const double factor) noexcept {
+            return factor * some_number_on_interface + volume_argument;
+          },
+          box, 2.);
+  CHECK(computed_number_on_interfaces.size() ==
+        expected_result_on_interfaces.size());
+  for (const auto& direction_and_expected_result :
+       expected_result_on_interfaces) {
+    CHECK(
+        computed_number_on_interfaces.at(direction_and_expected_result.first) ==
+        direction_and_expected_result.second);
+  }
+  /// [interface_apply_example]
+
+  // Test volume base tag
+  const auto computed_numbers_with_base_tag =
+      interface_apply<DirectionsTag, tmpl::list<SomeNumber, VolumeArgumentBase>,
+                      tmpl::list<VolumeArgumentBase>>(
+          [](const double& some_number_on_interface,
+             const double& volume_argument, const double factor) noexcept {
+            return factor * some_number_on_interface + volume_argument;
+          },
+          box, 2.);
+  CHECK(computed_numbers_with_base_tag == computed_number_on_interfaces);
 }
 
 SPECTRE_TEST_CASE("Unit.Domain.InterfaceHelpers", "[Unit][Domain]") {
-  {
-    const ElementId<1> element_id{0, {{{1, 0}}}};
-    const ElementId<1> east_id{0, {{{1, 1}}}};
-    const Element<1> element{element_id,
-                             {{Direction<1>::upper_xi(), {{east_id}, {}}}}};
-    test_interface_apply(element);
-  }
+  test_interface_apply<1, ::Tags::InternalDirections<1>>(
+      // Reference element has one internal direction:
+      // [ X | ]-> xi
+      {{0, {{{1, 0}}}}, {{Direction<1>::upper_xi(), {{{0, {{{1, 1}}}}}, {}}}}},
+      {{Direction<1>::upper_xi(), 2.}}, {{Direction<1>::upper_xi(), 5.}});
+  test_interface_apply<1, ::Tags::InternalDirections<1>>(
+      // Reference element no internal directions:
+      // [ | X | | ]-> xi
+      {{0, {{{2, 1}}}}, {}}, {}, {});
```

Comment:
Shouldn't no internal directions be `[ X ]` and `{0, 0}`?

Reply:
yes, fixed.